spec/Appendix B -- Grammar Summary.md

```diff
@@ -20,10 +26,10 @@ WhiteSpace ::
 
 LineTerminator ::
   - "New Line (U+000A)"
-  - "Carriage Return (U+000D)" [ lookahead ! "New Line (U+000A)" ]
+  - "Carriage Return (U+000D)" [lookahead != "New Line (U+000A)"]
```

Comment:
wouldn't be simpler to express a case like this by making the NewLine in the third case optional?

Reply:
Following up on this. It would actually create ambiguity this RFC is trying to solve.

Example:

```
LineTerminator ::
  - <NL>
  - <CR> <NL>?
```

Then how many LineTerminators are here: `<CR><NL>`? The answer could be 2 because we can apply the second rule `<CR> <NL>?` with the `<NL>?` skipped, and then apply the first rule `<NL>`.

However for the previously stated rules (below), then the answer *can only be 1* because the second rule `<CR> [!= <NL>]` cannot match, however the third rule `<CR> <NL>` does match, but consumes the whole sequence.

```
LineTerminator ::
  - <NL>
  - <CR> [!= <NL>]
  - <CR> <NL>
```